tests/datasets/test_dataset_factory.py

```diff
@@ -0,0 +1,85 @@
+import threading
+from typing import Iterator
+
+import pytest
+
+from snuba import settings
+from snuba.datasets.dataset import Dataset
+from snuba.datasets.factory import (
+    InvalidDatasetError,
+    get_dataset,
+    get_dataset_name,
+    get_enabled_dataset_names,
+)
+
+
+def test_get_dataset() -> None:
+    for ds_name in [
+        "discover",
+        "events",
+        "groupassignee",
+        "groupedmessage",
+        "metrics",
+        "outcomes",
+        "outcomes_raw",
+        "sessions",
+        "transactions",
+        "profiles",
+        "functions",
+        "generic_metrics",
+        "replays",
+    ]:
+        factory_ds = get_dataset(ds_name)
+        assert isinstance(factory_ds, Dataset)
+        assert get_dataset_name(factory_ds) == ds_name
+
+
+def test_get_dataset_multithreaded_collision() -> None:
```

Comment:
What happens if this test fails? Does the exception automatically get bubbled up?

Reply:
Good point, the exception was not automatically bubbled up to the main thread, only existed in child thread's context, and only writing error to stdout. Added some error handling to fix this and fail the test.